wolfcrypt/src/asn.c

```diff
@@ -464,6 +464,17 @@ static int GetASNInt(const byte* input, word32* inOutIdx, int* len,
         return ret;
 
     if (*len > 0) {
+
+#ifndef WOLFSSL_ASN_INT_LEAD_0_ANY
```

Comment:
Why does this here?
This function is a generic ASN.1 integer parsing function.
Negative numbers are allowed, just not in X.509 serial numbers.

Reply:
It's only adding X.690-pursuant rejection of frivolously padded negative numbers, just as frivolously padded positive numbers were already rejected.  In both cases (as before in the case of positive), defining `WOLFSSL_ASN_INT_LEAD_0_ANY` inhibits the rejection.

More to the point: negative numbers are still allowed by the routine as modified, as they should be.